Tidy up Documentation to reflect current status (#47)
veraison/docs: project-overview.md

### Tokens

Many real world attestation solutions and potential extrapolations of their use have been considered to ensure that the general logic within the components performs the correct set of operations to provide support for the maximum set of deployments. Developing understanding and modelling the necessary flows for their verification process has driven the Veraison design.

> To ensure that the code backs up the design, reference support for multiple attestation tokens are supported in the project. The following are currently supported:

Reviewer: ```suggestion
To ensure that the code backs up the design, reference support for multiple attestation evidence formats are supported in the project. The following are currently supported:
```

Author: The title is tokens for the section and we have used attestation tokens so I would prefer to retain this terminology!

Reviewer: as pointed out by Sergei in another review comment, some of these (namely DICE) are not strictly "tokens"

Author: This has been addressed in the titles and clarified on the individual specific bullets

> * EAT Tokens, with the ARM PSA Profiles (Profile 1 & Profile 2) being a specific test set.

Reviewer: ```suggestion
* EAT Tokens, with the [ARM PSA Profiles](https://datatracker.ietf.org/doc/draft-tschofenig-rats-psa-token/) being a specific test set.
```

> * ARM Confidential Computing Architecture (CCA) Token and its variant [Parsec-CCA](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-cca.md)

Reviewer: _variant_ is probably not the right word (there's no variation), _composition_ maybe more correct

Author: A variant is another version of something. This is correct usage, as Parsec tpm is a variant of base CCA token

Reviewer: hmm, the CCA token is reused as-is by Parsec CCA, so there is absolutely no variation.  It is a composition of CCA and another piece that's stacked on top.

Author: ok, without being hung up on the term, I have made the change!

> * ARM Confidential Computing Architecture (CCA) Token and its composition [Parsec-CCA](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-cca.md)

Reviewer: ```suggestion
* ARM Confidential Computing Architecture (CCA) Token and its composition [Parsec-CCA](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-cca.md).
```

> * [DICE Tokens](https://trustedcomputinggroup.org/work-groups/dice-architectures/) (note: this does not implement any specific DICE architecture). If you're planning any specific profile of DICE protocol implementations, we would be very interested in discussing them.

Reviewer: not sure what is meant by "DICE architecture" in this context?  In theory we can do both TCG and Open.

Author: Have addressed the concern and corrected some text, please have a look

> * [DICE Tokens](https://trustedcomputinggroup.org/work-groups/dice-architectures/) (note: this does not implement any specific DICE architecture). If you're planning any specific profile of DICE protocol implementations, we would be very interested in discussing them.

Reviewer 2: ```suggestion
* [DICE](https://trustedcomputinggroup.org/work-groups/dice-architectures/) based attestation tokens (note: this does not adhere to any specific DICE-based protocol). If you're planning any particular DICE implementations, we would be very interested in discussing them.
```

Should probably also remove the reference to "DICE Tokens" here -- DICE specifically doesn't define anything like a token format, so there no such thing as a "DICE Token". (just like there so no such thing as a "TPM Token").

For the same reason, the term "profile" isn't really applicable here.

> * [DICE layered Evidence](https://trustedcomputinggroup.org/work-groups/dice-architectures/) (note: this does not adhere to any specific DICE-based protocol). If you're planning any particular DICE protocol implementation, we would be very interested in discussing it.

Reviewer: ```suggestion
* [DICE layered Evidence](https://trustedcomputinggroup.org/work-groups/dice-architectures/).  Note that this does not adhere to any specific DICE-based protocol.  If you're planning any particular DICE protocol implementation, we would be very interested in discussing it.
```

Author: Again a editorial view of looking at the same thing, not much different from the text which is currently written. I would retain an explicit note as this does better on emphasis.

Reviewer: I think the proposal reads a bit better.  Removing the parenthetical connects the two related sentences ("Note that [...]." and "If you are planning [...]) and makes the flow smoother.  Besides, "note" should be capitalised.

> * TPM based attestation tokens namely [Enactrust](https://enacttrust.com/)  & [Parsec-TPM](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-tpm.md)

Reviewer: * typo "Enactrust"
* extra space before ampersand
* missing comma after "tokens"

Author: Done!

> * TPM based attestation tokens, namely [Enacttrust](https://enacttrust.com/) & [Parsec-TPM](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-tpm.md)

Reviewer: ```suggestion
* TPM-based attestation evidence formats, namely [EnactTrust](https://enacttrust.com/) & [Parsec-TPM](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-tpm.md).
```

Author: again no need to change to Evidence formats

Reviewer: this goes along with the other

### Tokens

Many real world attestation solutions and potential extrapolations of their use have been considered to ensure that the general logic within the components performs the correct set of operations to provide support for the maximum set of deployments. Developing understanding and modelling the necessary flows for their verification process has driven the Veraison design.
To ensure that the code backs up the design, reference support for multiple attestation tokens are supported in the project. The following are currently supported:

*  EAT Tokens, with the [ARM PSA Profiles](https://datatracker.ietf.org/doc/draft-tschofenig-rats-psa-token/) being a specific test set.
* ARM Confidential Computing Architecture (CCA) Token and its composition [Parsec-CCA](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-cca.md).

> * [DICE layered Evidence](https://trustedcomputinggroup.org/work-groups/dice-architectures/) (Note: this does not adhere to any specific DICE-based protocol). If you're planning any particular DICE protocol implementation, we would be very interested in discussing it.
> * TPM based attestation tokens, namely [Enacttrust](https://enacttrust.com/) & [Parsec-TPM](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-tpm.md).

Reviewer: * moving the closing parenthesis to the right place
* correct capitalisation of EnactTrust

```suggestion
* [DICE layered Evidence](https://trustedcomputinggroup.org/work-groups/dice-architectures/) (Note: this does not adhere to any specific DICE-based protocol. If you're planning any particular DICE protocol implementation, we would be very interested in discussing it.)
* TPM based attestation tokens, namely [EnactTrust](https://enacttrust.com/) & [Parsec-TPM](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-tpm.md).
```

Author: The parenthesis is really meant for the Note, not for the second statement. 
Thanks for the suggestion on the EnactTrust, will do it!

Author: ok, thinking a bit more, it does not hurt to bring everything in the parenthesis, I will change this!

Reviewer: The second sentence is actually part of the note.  See the original text please.

Author: > The second sentence is actually part of the note. See the original text please.

yes, i have corrected it already! Thanks!

To ensure that the code backs up the design, reference support for multiple attestation tokens are supported in the project. The following are currently supported:

* EAT Tokens, with the ARM PSA Profiles (Profile 1 & Profile 2) being a specific test set.
* ARM Confidential Computing Architecture (CCA) Token and its composition [Parsec-CCA](https://github.com/CCC-Attestation/attested-tls-poc/blob/main/doc/parsec-evidence-cca.md)

> * [DICE layered Evidence](https://trustedcomputinggroup.org/work-groups/dice-architectures/) (note: this does not adhere to any specific DICE-based protocol). If you're planning any particular of DICE protocol implementations, we would be very interested in discussing them.

Author: This needs a bit of tidy up, need some adjustment : will do

> * [DICE layered Evidence](https://trustedcomputinggroup.org/work-groups/dice-architectures/) (note: this does not adhere to any specific DICE-based protocol). If you're planning any particular of DICE protocol implementations, we would be very interested in discussing them.

Author: Yes, I had corrected it in the latest commit. Thanks.
